psycopg3/transaction.py

```diff
@@ -86,14 +89,15 @@ def __exit__(
         exc_type: Optional[Type[BaseException]],
         exc_val: Optional[BaseException],
         exc_tb: Optional[TracebackType],
-    ) -> None:
+    ) -> bool:
```

Comment:
Why does this return bool?

Reply:
`__exit__` returns a boolean to signal whether the exception should be propagated or not, per the [Python Data Model](https://docs.python.org/3/reference/datamodel.html#object.__exit__) -- so that we can swallow the `Rollback()` exception, where appropriate.

Did I misunderstand the question?

Reply:
No it's me who didn't know the behaviour of returning True from __exit__ :disappointed: This is all ok (as long as mypy is ok with all the possible exit paths)